wordpress-mobile/WordPress-Android
# Hide "Check email" button in the signup magic link screen when email app is not available

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/15883

#### Related PR: https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/77

### Description

This PR checks the availability of the default email app prior to displaying the "Check email" button, and avoids displaying it if we know the email app is not available via query. In that case, the button would only result in a toast error message, so we avoid that error by merely omitting the button. The instructions on the screen already describe to the user to open the email on that device to use the magic signup link, and removing the button makes it more clear that this step must be performed manually.

#### Update

This PR also fixes email app queries for API 30 by declaring the query intent in the manifest.

To test:

---

**Precondition**: No default email app

To establish the precondition, it was necessary in my case (on an emulator on API 29) to _disable_ the preinstalled gmail app. In other cases / devices, it may be sufficient to clear the email app(s) as the default app for email.

1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide a fake email (e.g. "fake1234567@example.com")
3. On the next screen notice the absence of the `Check email` button

**Screenshot**:
||
|-|
|<img src="https://user-images.githubusercontent.com/8507675/152961891-e5c6b35b-87e6-48f1-a4eb-e398770d4f3a.png" width="360">|

---

**Precondition**: With a default email app

Set the email (gmail, or other) as the default app for email.

1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide a fake email (e.g. "fake1234567@example.com")
3. On the next screen, check the presence of the `Check email` button
4. Press the `Check email` button
5. The default email client app should open on the device

**Screenshot**:
||
|-|
|<img src="https://user-images.githubusercontent.com/8507675/152961906-b6195193-4713-4154-95fe-b685022dfe85.png" width="360">|

---

## Regression Notes
1. Potential unintended areas of impact
Other usages / consumers of the signup flow in the login library.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Existing signup magic link unit tests.

3. What automated tests I added (or what prevented me from doing so)
I attempted to add tests in the library repository, however it seems that the fragment under test is injected (dagger), making `FragmentScenario` unusable there. I also explored instrumentation tests for that UI in the library, however the footprint for that was very large and out of scope for this PR. I also explored adding instrumentation tests in WordPress-Android, however it was not clear if / how to reliably establish the necessary preconditions (i.e. that `packageManager.queryIntentActivities` would return either an empty list or an email app).

PR submission checklist:
